gadget/gadget-amd64.yaml

```diff
@@ -25,12 +25,22 @@ volumes:
         type: C12A7328-F81F-11D2-BA4B-00A0C93EC93B
         size: 1200M
         update:
-          edition: 2
+          edition: 3
         content:
           - source: grubx64.efi
-            target: EFI/boot/grubx64.efi
+            target: EFI/BOOT/grubx64.efi
+          - source: grubx64.efi
+            target: EFI/ubuntu/grubx64.efi
+          - source: shim.efi.signed
+            target: EFI/BOOT/BOOTX64.efi
+          - source: fbx64.efi
+            target: EFI/BOOT/fbx64.efi
+          - source: mmx64.efi
```

Comment:
I think the mok manager should also go in `EFI\ubuntu`.

Reply:
Sounds good. I don't really know what the mok manager is used for, and there was some consternation about adding it in the past: https://github.com/snapcore/pc-gadget/pull/56#discussion_r815868895

It's not one of the boot assets added by the snapd PR #13205 (https://github.com/snapcore/snapd/pull/13205/files#diff-dca334c34d6acf0ce9fb49a95fb836caafe064f7fd2eda43a2185d6d9c7e20e5R464-R476), so I'm not sure if it's something we actually need to add.

What do you think?

Reply:
I would love to have it. The mok keys can be used to sign your own kernels. But it can also be used by anything that uses the secondary trusted keyring in the kernel: kernel modules, signed dm-verity, etc.

Also there is no security argument here, because there is a signed efi that anyone can install manually. We are just making it more annoying to install it for legitimate users.

If it causes some issues and would delay this PR, you do not have to add it.

But for sure, if we add it in `EFI\BOOT`, it should also go in `EFI\ubuntu`. It feels very weird to have it in only one. That would be, you can install machine owner keys, but only if in the bios, you force to boot the default efi bootloader (`EFI\BOOT\bootx64.efi`) instead of the configured `BootOrder`.

To sum up. Either add `EFI\ubuntu\mmx64.efi` or remove `EFI\BOOT\mmx64.efi`.

Reply:
Probably we can live with MokManager only in `EFI\ubuntu`, as we have seen that grub will not ever run from `EFI\boot` if we have fbx64.efi. As MM is called either when shim tries to run grub but it has not a valid signature or when the OS modifies EFI variables related to the MoK - and the OS is expected to have been started by grub in `EFI\ubuntu`.

Reply:
Sounds correct.

Reply:
Makes sense. I've removed it now, thanks!